libraries/ESP8266LLMNR/ESP8266LLMNR.cpp

```diff
@@ -0,0 +1,284 @@
+ * Copyright (C) 2017 Stephen Warren <swarren@wwwdotorg.org>
+ *
+ * Based on:
+ * ESP8266 Multicast DNS (port of CC3000 Multicast DNS library)
+ * Version 1.1
+ * Copyright (c) 2013 Tony DiCola (tony@tonydicola.com)
+ * ESP8266 port (c) 2015 Ivan Grokhotkov (ivan@esp8266.com)
+ * MDNS-SD Suport 2015 Hristo Gochkov
+ * Extended MDNS-SD support 2016 Lars Englund (lars.englund@gmail.com)
+ *
+ * License (MIT license):
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * Reference:
+ * https://tools.ietf.org/html/rfc4795 (LLMNR)
+ * https://tools.ietf.org/html/rfc1035 (DNS)
+ */
+
+#include <debug.h>
+#include <functional>
+#include <ESP8266LLMNR.h>
+#include <ESP8266WiFi.h>
+#include <WiFiUdp.h>
+
+extern "C" {
+#include <user_interface.h>
+}
+
+#include <lwip/udp.h>
+#include <lwip/igmp.h>
+#include <include/UdpContext.h>
+
+//#define LLMNR_DEBUG
+
+#define BIT(x) (1 << (x))
+
+#define FLAGS_QR                BIT(15)
+#define FLAGS_OP_SHIFT          11
+#define FLAGS_OP_MASK           0xf
+#define FLAGS_C                 BIT(10)
+#define FLAGS_TC                BIT(9)
+#define FLAGS_T                 BIT(8)
+#define FLAGS_RCODE_SHIFT       0
+#define FLAGS_RCODE_MASK        0xf
+
+#define _conn_read16() (((uint16_t)_conn->read() << 8) | _conn->read())
+#define _conn_read8() _conn->read()
+#define _conn_readS(b, l) _conn->read((b), (l));
+
+static const IPAddress LLMNR_MULTICAST_ADDR(224, 0, 0, 252);
+static const int LLMNR_MULTICAST_TTL = 1;
+static const int LLMNR_PORT = 5355;
+
+LLMNRResponder::LLMNRResponder() :
+    _conn(0) {
+}
+
+LLMNRResponder::~LLMNRResponder() {
+    if (_conn) {
+        _conn->unref();
+    }
+}
+
+bool LLMNRResponder::begin(const char* hostname) {
+    // Max length for a single label in DNS
+    if (strlen(hostname) > 63) {
+        return false;
+    }
+
+    _hostname = hostname;
+    _hostname.toLowerCase();
+
+    WiFi.onStationModeGotIP([this](const WiFiEventStationModeGotIP& event){
```

Comment:
You need to save the return value as a member variable of the class. The callback lifetime is controlled by the returned `EventHandler` value.
Same thing a few lines below.